Changes our map_format to SIDE_MAP

LemonInTheDark opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This does nothing currently, but will allow me to test for layering issues on LIVE, rather then in just wallening.
Oh also I'm packaging in a fix to one of my macros that I wrote wrong, as a joke

[removes SEE_BLACKNESS usage, because we actually cannot use it effectively](https://github.com/tgstation/tgstation/pull/70162/commits/c9a19dd7cce95038340ebd5c1a6e4cb27ee7c9ee)
[c9a19dd](https://github.com/tgstation/tgstation/pull/70162/commits/c9a19dd7cce95038340ebd5c1a6e4cb27ee7c9ee)

Sidemap removes the ability to control it on a plane, so it basically just means there's an uncontrollable black slate even if you have other toggles set.

This just like, removes that, since it's silly

[fixes weird layering on solars and ai portraits. Pixel y was casuing things to render below who shouldn't](https://github.com/tgstation/tgstation/pull/70162/commits/3885b9d9ed634cdc4c8041b19df5b5ea9a1a37ae)
[3885b9d](https://github.com/tgstation/tgstation/pull/70162/commits/3885b9d9ed634cdc4c8041b19df5b5ea9a1a37ae)

[Fixes flicker issues](https://github.com/tgstation/tgstation/pull/70162/commits/2defc0ad20a0ee7d12e0b071f6d31b6127b8765d)
[2defc0a](https://github.com/tgstation/tgstation/pull/70162/commits/2defc0ad20a0ee7d12e0b071f6d31b6127b8765d)

Offsetting the vis_contents'd objects down physically, and then up visually resolves the confliciting that was going on between the text and its display.

This resolves the existing reported flickering issues

[fixes plated food not appearing in world](https://github.com/tgstation/tgstation/pull/70162/commits/28a34c64f830660d7fb1cc669b9fc3ed9f5c7d61)
[28a34c6](https://github.com/tgstation/tgstation/pull/70162/commits/28a34c64f830660d7fb1cc669b9fc3ed9f5c7d61)

pixel_y'd vis_contents strikes again. It's a tad hacky but we'll just use pixel_z for this

[Adds wall and upper wall plane masters](https://github.com/tgstation/tgstation/pull/70162/commits/89fe2b4eb40edc36879e4e1954dee8616be94522)
[89fe2b4](https://github.com/tgstation/tgstation/pull/70162/commits/89fe2b4eb40edc36879e4e1954dee8616be94522)

We use these + the floor and space planes to build a mask of all the visible turfs.
Then we take that, stick it in a plane master, and mask the emissive plane with it.

This solves the lighting fulldark screen object getting cut by emissives
Shifts some planes around to match this new layering. Also ensures we only shift fullscreen objects if they don't object to it.

[compresses plane master controllers](https://github.com/tgstation/tgstation/pull/70162/commits/bd64cc196a4265d42809eebbd1afa46fa33a576d)
[bd64cc1](https://github.com/tgstation/tgstation/pull/70162/commits/bd64cc196a4265d42809eebbd1afa46fa33a576d)

we don't use them for much rn, but we might in future so I'm keeping it as a convienince thing

:cl:
refactor: The logic of how we well, render things has changed. Make an issue report if anything looks funky, particularly layers. PLEASE USE YOUR EYES
/:cl:

LemonInTheDark committed:
Changes our map_format to SIDE_MAP

This does nothing currently, but will allow me to test for layering
issues on LIVE, rather then in just wallening.

Oh also I'm packaging in a fix to one of my macros that I wrote wrong,
as a joke

LemonInTheDark commented:
I swear I'll document this later, when we actually rely on it

axietheaxolotl commented:
the wallening will never happen - oranges, 2022

LemonInTheDark commented:
> wallening by end of 2022, potatoe promised me"
> \-oranges

Gotta keep my job man

Imaginos16 commented:
Lets go????

Mothblocks commented:
remove the dnm when you're confident

msgerbs commented:
Does this actually change things? I'm looking at the byond docs and it's not clear to me what this actually changes.

LemonInTheDark commented:
This is built to break. I want to know what's iffy about side_map, and the only way to do that is farm bugs like the ones that spawned #58051
The trouble is people are bad at reporting things like that, so the best way I can think of of accomplishing it is just getting it fullmerged.
If that isn't acceptable I'll mass tm it, not like people edit world.dm all that often

AndrewL97 commented:
Hey @LemonInTheDark, as the shadow-host for tg, this has my thumbs up
The other citrus will back me up on this

msgerbs commented:
Yeah that. Not saying not to merge it, not that anyone would listen if I did, I am just genuinely curious how this affects rendering. Byond's docs imply you need to manually modify your tile sizes and layering anyway with this, but doesn't really say how this var affects anything.

Tastyfish commented:
> Yeah that. Not saying not to merge it, not that anyone would listen if I did, I am just genuinely curious how this affects rendering. Byond's docs imply you need to manually modify your tile sizes and layering anyway with this, but doesn't really say how this var affects anything.

Y position takes precedence over layer.
> [SIDE_MAP] looks very similar to a top-down view but it is intended for more of a 3/4 perspective, where tiles lower on the screen are considered closer to the viewer. Because this impacts the way layers work, most of the layering behavior is the same as with isometric.

> In isometric, tiles that are nearer to the viewer's perspective are drawn in front of tiles farther back, regardless of layer. Layers only count within an individual tile. This means that if you want to have a vertical structure "stick up" to partially hide something behind it, the icon sticking up should always be on a tile forward from the one being partly covered. E.g. if you have a wall taking up part of your tile, it needs to be at the "back" end of the tile to properly hide anything on the tiles behind it.

Mothblocks commented:
@LemonInTheDark How is this going

LemonInTheDark commented:
I know I want it, but I need to write up documentation on it. Will do soon

Mothblocks commented:
Why is the bot not reporting any test merges in the past 200 rounds You are test merging this right Right

LemonInTheDark commented:
No? I uh, I think I may have miscommunicated

Mothblocks commented:
Test merging this would be smart

LemonInTheDark commented:
yeah yeah

Mothblocks commented:
we have 102 open prs lemon

LemonInTheDark commented:
Listen my plan was to just merge this and eat the bug reports.
I did actually TM it tho, up on sybil now.

LemonInTheDark committed:
Removes SEE_BLACKNESS usage, because we actually cannot use it effectively

Sidemap removes the ability to control it on a plane, so it basically
just means there's an uncontrollable black slate even if you have other
toggles set.

This just like, removes that, since it's silly

Mothblocks commented on code/_onclick/hud/rendering/_render_readme.md:
```diff
@@ -20,7 +20,7 @@ B) Use the undocumented debug printer which reads of data on icons rendering, th
 
 1. Transforms are very slow on clientside. This is not usually noticable, but if you start using large amounts of them it will grind you to a halt quickly, regardless of whether its on overlays or objs
-2. The darkness plane. The darkness plane has specific variables it needs to render correctly, and these can be found in the plane masters file. it is composed internally of two parts, a black mask over the clients screen, and a non rendering mask that blocks all luminosity=0 turfs and their contents from rendering if the SEE_BLACKNESS flag is set properly. It behaves very oddly, such as forcing  ...
+2. The darkness plane. The darkness plane has specific variables it needs to render correctly, and these can be found in the plane masters file. it is composed internally of two parts, a black mask over the clients screen, and a non rendering mask that blocks all luminosity=0 turfs and their contents from rendering if the SEE_BLACKNESS flag is set properly. The blocker will always block rendering ...
```
```suggestion
2. The darkness plane. This is unused, as it doesn't work with our rendering format, so this section is purely academic. The darkness plane has specific variables it needs to render correctly, and these can be found in the plane masters file. it is composed internally of two parts, a black mask over the clients screen, and a non rendering mask that blocks all luminosity=0 turfs and their contents from rendering if the SEE_BLACKNESS flag is set properly. The blocker will always block rendering but the mask can be layered under other objects.
```

nit, should probably let people know faster

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into roll-the-old-chariot-along

LemonInTheDark commented:
testmerging again

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into roll-the-old-chariot-along

LemonInTheDark committed:
fixes weird layering on solars and ai portraits. Pixel y was casuing things to render below who shouldn't

LemonInTheDark committed:
Fixes flicker issues

Offsetting the vis_contents'd objects down physically, and then up visually
resolves the confliciting that was going on between the text and its
display.

This resolves the existing reported flickering issues

LemonInTheDark committed:
Moth suggestion

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

LemonInTheDark commented:
Retestmerged, hopefully I can farm some more issues

Tastyfish commented:
For the cases where you're doing the pixel_z pixel_y trick, make sure they update on direction changing (setDir or the signal), or they'll visually break if used on a shuttle.

LemonInTheDark commented:
It's only really used for overlays/vis_contents objects, so it ought to be fine. there might be an issue of things layering as they shouldn't tho, not totally sure what to do about that might just have to bully lummox. we'll see

Tastyfish commented:
> It's only really used for overlays/vis_contents objects, so it ought to be fine. there might be an issue of things layering as they shouldn't tho, not totally sure what to do about that might just have to bully lummox. we'll see

Alrightyt

LemonInTheDark committed:
fixes plated food not appearing in world. pixel_y'd vis_contents strikes again. It's a tad hacky but we'll just use pixel_z for this

LemonInTheDark committed:
Adds wall and upper wall plane masters

We use these + the floor and space planes to build a mask of all the
visible turfs.
Then we take that, stick it in a plane master, and mask the emissive
plane with it.

This solves the lighting fulldark screen object getting cut by emissives
Shifts some planes around to match this new layering. Also ensures we
only shift fullscreen objects if they don't object to it.

LemonInTheDark committed:
compresses plane master controllers. we don't use them for much rn, but we might in future so I'm keeping it as a convienince thing

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into roll-the-old-chariot-along

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into roll-the-old-chariot-along

LemonInTheDark commented:
Seen no bugs in a week, at least that have been linked to this pr. I don't think I'll get much more out of this.
Will write docs for sidemap in a day or two, then this is ready for merge

Mothblocks commented:
just write them separately

optimumtact commented:
that is a very good way to not get any written at all
